resizable-polygon/demo.js

```diff
@@ -0,0 +1,139 @@
+      '<circle cx="10" cy="10" r="7" fill="transparent" stroke="red" stroke-width="4"/>' +
+      '</svg>',
+      polygon = new H.map.Polygon(
+        new H.geo.Polygon(new H.geo.LineString([51.2, 21.51, 0, 51.2, 25.6, 0, 49.2, 25.9, 0, 48.7, 22.5, 0, 49.9, 24, 0, 50.5, 24, 0])),
+        {
+          style: {fillColor: 'rgba(150, 100, 0, .8)', lineWidth: 0}
+        }
+      ),
+      polygonGroup = new H.map.Group({
+        volatility: true, // mark the group as volatile for smooth dragging of all it's objects
+        objects: [polygon]
+      }),
+      polygonTimeout;
+
+  // ensure that the polygon can receive drag events
+  polygon.draggable = true;
+
+  // create markers for each polygon's node which will be used for dragging
+  polygon.getGeometry().getExterior().eachLatLngAlt(function(lat, lng, alt, index) {
+    var marker = new H.map.Marker(
+      {lat, lng},
+      {
+        icon: new H.map.Icon(svgCircle, {anchor: {x: 10, y: 10}})
+      }
+    );
+    marker.setVisibility(false);
+    marker.draggable = true;
+    marker.setData({'nodeIndex': index})
+    polygonGroup.addObject(marker);
+  });
+
+  // add group with polygon and it's nodes (markers)
+  map.addObject(polygonGroup);
+
+  // event listener for polygon group to show markers if moved in with mouse (or touched on touch devices)
+  polygonGroup.addEventListener('pointerenter', function(evt) {
+    var group = this;
+    if (polygonTimeout) {
+      clearTimeout(polygonTimeout);
+      polygonTimeout = null;
+    }
+
+    // show markers
+    group.forEach(function(object) {
+      if (object instanceof H.map.Marker) {
+        object.setVisibility(true);
+      }
+    });
+  }, true);
+
+  // event listener for polygon group to hide markers if moved out with mouse (or released finger on touch devices)
+  // the outline is hidden on touch devices after specific timeout
+  polygonGroup.addEventListener('pointerleave', function(evt) {
+    var group = this,
+        timeout = (evt.currentPointer.type == 'touch') ? 1000 : 0;
+
+    polygonTimeout = setTimeout(function() {
+      group.forEach(function(object) {
+        if (object instanceof H.map.Marker) {
+          object.setVisibility(false);
+        }
+      });
+    }, timeout);
+
+    document.body.style.cursor = 'default';
+  }, true);
+
+  // event listener for polygon group to change the cursor if mouse position is over markers
+  polygonGroup.addEventListener('pointermove', function(evt) {
+    if (evt.target instanceof H.map.Marker) {
+      document.body.style.cursor = 'pointer';
+    } else {
+      document.body.style.cursor = 'default'
+    }
+  }, true);
+
+  // event listener for polygon group to resize the geo polygon object if dragging over markers
+  polygonGroup.addEventListener('drag', function(evt) {
+    var pointer = evt.currentPointer,
+        group = this,
+        polygon = group.getObjects()[0],
```

Comment:
I'd suggest to use the variable from the outer scope, and remove the "this" assignement